### PR TITLE
docs: Added `-GNinja` to cmake configuration commands

### PR DESCRIPTION
**Pull request by tmatsuo**
docs: Also fixed the directory name for clang example

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10532)

### CODE REVIEW

File: doc/contributor/howto-guide-setup-cmake-environment.md [hunk 1]
```diff
@@ -60,7 +60,8 @@ to the `cmake` configure step:
 
 ```shell
 cd $HOME/google-cloud-cpp
-cmake -H. -Bcmake-out/home -DCMAKE_TOOLCHAIN_FILE=$HOME/vcpkg/scripts/buildsystems/vcpkg.cmake
+cmake -H. -GNinja -Bcmake-out/home \
```
Comment (coryan) [inline doc/contributor/howto-guide-setup-cmake-environment.md:63]: `-H` is an undocumented option. With CMake >= 3.10 (which we now require) we can use the documented `-S .` option (note the space).  Please fix that too.

Comment (coryan) [reply doc/contributor/howto-guide-setup-cmake-environment.md:63]: optional: in scripts we would sort these options differently we generally use: `cmake -G Ninja -S . -B cmake-out/home`.

Comment (coryan) [inline doc/contributor/howto-guide-setup-cmake-environment.md:63]: Since we wrote this, most CI scripts write to `build-out/*`, maybe do that here too?

**Review by coryan — CHANGES_REQUESTED** [review]

File: doc/contributor/howto-guide-setup-cmake-environment.md (hunk 1 above)
Comment (tmatsuo) [reply doc/contributor/howto-guide-setup-cmake-environment.md]: Done

Comment (tmatsuo) [reply doc/contributor/howto-guide-setup-cmake-environment.md]: Done

**Comment by tmatsuo** [comment]
@coryan Thanks, PTAL

**Review by coryan — APPROVED** [review]

**Pull request merged**